src/java/org/apache/cassandra/sensors/RequestSensorsFactory.java

```diff
@@ -44,4 +49,32 @@ default RequestSensors create(String keyspace)
     {
         return NoOpRequestSensors.instance;
     }
+
+    /**
+     * Returns a request sensor to string encoder that will be used to encode sensor as a custom header in the internode message {@link Message.Header#customParams()} bytes map.
+     * Request sensors track usage per request. See {@link RequestSensors}
+     * Implemenations should:
+     * <li> memoize their encoders and make them very efficient as encoders are potentially invoked on each request</li>
+     * <li> encode enough information to differentiate between sensors of the same type that belong to the same request but different keyspaces and/or tables. </li>
+     *
+     * @return a function that encodes a sensor to a string
+     */
+    default Function<Sensor, String> requestSensorEncoder()
```

Comment:
Alternatively, I can combine `requestSensorEncoder` &  `registrySensorEncoder` into one by either:
* `Function<Sensor, String> createSensorEncoder(Scope scope)` where scope is an `Enum` of `{REQUEST, TABLE}`. 
* Create a separate interface `SensorEncoder` with a create method that accepts both `Sensor` and `Scope`

Reply:
https://github.com/datastax/cassandra/pull/1295/commits/31dfa8f4e6d2ede8e9e7177a569136ded28c3959